### ISSUE TITLE
The query contains the 'left Join' operation and 'ROW_number' operation, and the results are inconsistent after multiple executions.

### ISSUE THREAD

**Issue by tonyDong-code** [StarRocks/starrocks#9558]
### Steps to reproduce the behavior (Required)

```sql
CREATE TABLE `tmp_info` (
  `var_id` int(11) NOT NULL COMMENT "var",
  `var_name` varchar(65533) NOT NULL COMMENT "var",
  `interface_id` varchar(65533) NULL COMMENT "var",
  `update_time` datetime NOT NULL COMMENT "var"
) ENGINE=OLAP 
UNIQUE KEY(`var_id`)
COMMENT "var"
DISTRIBUTED BY HASH(`var_id`) BUCKETS 3 
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
);
```

```sql
insert into tmp_info values 
(12963,'var','var','2022-08-04 11:23:39.280048'),
(12997,'var','var','2022-08-04 11:23:39.292017');
```

```sql
SELECT a.*, b.var_name AS varname
FROM ( 
    select 12963 as var_id 
    union 
    select 12997 as var_id
) a
left outer JOIN (
    select
        var_id,
        var_name,
        update_time,
        interface_id
    from (
        SELECT 
            ROW_NUMBER() over(PARTITION by var_id order by update_time, var_name) as rk,
            var_id,
            var_name,
            update_time,
            interface_id
        FROM tmp_info
    ) t1
    where t1.rk = 1 
) b
ON a.var_id = b.var_id;
```

### Expected behavior (Required)
![图片](https://user-images.githubusercontent.com/72612520/182761649-ca7eaba7-e644-4942-8670-bebbc60e2f44.png)

### Real behavior (Required)
The above query is repeated several times in a row, yielding several different results:
![图片](https://user-images.githubusercontent.com/72612520/182761674-bfc8f921-f71b-4f74-98ed-8996a5a601c8.png)
![图片](https://user-images.githubusercontent.com/72612520/182761689-22dc5338-116d-4247-8000-132c0ca3719f.png)
![图片](https://user-images.githubusercontent.com/72612520/182761726-a55e7963-f683-40ec-bfdd-cf51c6418a53.png)
![图片](https://user-images.githubusercontent.com/72612520/182761763-43a42d72-0a05-4049-8f8f-2eb35ff86fe5.png)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 
![图片](https://user-images.githubusercontent.com/72612520/182761906-c8b54a42-8ef8-4b3c-ad1b-1958359c458f.png)

The results of 'show variables'
![图片](https://user-images.githubusercontent.com/72612520/182762197-9abc100c-1456-4e77-8974-b57547112f7b.png)

![图片](https://user-images.githubusercontent.com/72612520/182762173-f67b31d0-5ac6-4676-8f7c-e64c9943d371.png)

![图片](https://user-images.githubusercontent.com/72612520/182762167-7cfa6076-55af-4ea9-89e5-a793f4c728c7.png)

**Comment by liuyehcf** [comment]
```sql
SELECT a.*, b.var_name AS varname, rk
FROM ( 
    SELECT var_id FROM tmp_info where var_id in(12963, 12997)
) a
left outer JOIN (
    select
        var_id,
        var_name,
        update_time,
        interface_id,
        rk
    from (
        SELECT 
            ROW_NUMBER() over(PARTITION by var_id order by update_time, var_name) as rk,
            var_id,
            var_name,
            update_time,
            interface_id
        FROM tmp_info where var_id in(12963, 12997)
    ) t1
) b
ON a.var_id = b.var_id;
```

**Comment by liuyehcf** [comment]
```sql
SELECT a.*, b.var_name AS varname, rk
FROM ( 
    select 12963 as var_id 
    union 
    select 12997 as var_id
) a
left outer JOIN (
    SELECT 
    ROW_NUMBER() over(PARTITION by var_id order by update_time, var_name) as rk,
    var_id,
    var_name,
    update_time,
    interface_id
    FROM tmp_info 
    where var_id in(12963, 12997)
) b
ON a.var_id = b.var_id;
```

**Comment by liuyehcf** [comment]
Introduced by #5541

**Comment by liuyehcf** [comment]
This issue will be solved in version `2.3.1` or later
